### DISCUSSION TITLE
Possibility to add own CSS code to modoboa

### DISCUSSION THREAD

**Discussion by piwats** [modoboa/modoboa#629] [Ideas]
Is it possible to add/inject own css code in Modoboa?

-Paul

**Comment by tonioo** [comment]
There is no official way to do this atm.

**Comment by piwats** [comment]
Is it possible to have f.e a way to point to a .css file somewhere in the settings/settings-file?

**Comment by tonioo** [comment]
What kind of modification would you like to do ?

**Comment by piwats** [comment]
Not much and nothing ground breaking. Change the background of the login page, maybe the font style/family, add shadow to the login box.

If you could add a hook and an option in the settings file, that would be great.
This is not a priority wish, but could be useful....

**Comment by eberkund** [comment]
Perhaps the ability to install additional themes could be added, a lot of software has this functionality. However IMO this is not very high priority for email server management software.

**Comment by ogg1980** [comment]
Meanwhile you can use nginx [sub_filter module](http://nginx.org/en/docs/http/ngx_http_sub_module.html) to get around this.

I use the following to inject custom CSS and add some tags on the login page (if you wan't multiple sub_filter per location, you need nginx 1.9.4 or later. Earlier versions are limited to one per location block.):

```
location / {
    include uwsgi_params;
    uwsgi_param UWSGI_SCRIPT instance.wsgi:application;
    uwsgi_pass modoboa;
    sub_filter '<link rel="stylesheet" type="text/css" href="/sitestatic/css/login.css" />'
        '<link rel="stylesheet" type="text/css" href="/sitestatic/css/login.css" />
<link rel="stylesheet" type="text/css" href="//domain.nx/mailadmin_login.css"/>';
    sub_filter '</form>'
    '</form><a href="http://modoboa.org/"><img src="//domain.nx/modoboa.svg" class="modoboa_logo"></a>';

    sub_filter_once on;
}

location ~ ^/(admin|stats|quarantine|core|user|webmail|sfilters)/ {
    include uwsgi_params;
    uwsgi_param UWSGI_SCRIPT instance.wsgi:application;
    uwsgi_pass modoboa;
    sub_filter '</head>'
'<link rel="stylesheet" type="text/css" href="//domain.nx/mailadmin.css"/>
</head>';

    sub_filter_once on;
}
```

If you use Apache, they have [mod_substitute](https://httpd.apache.org/docs/2.4/mod/mod_substitute.html) which does the same thing.

**Comment by LizAinslie** [comment]
this would be nice to have, but i would like a little bit more than just custom css. it would be nice if we could specify a custom template overrides directory. Main reason I'd like this is so I can use something like [fomantic-ui](https://fomantic-ui.com) instead of Bootstrap
